### PR TITLE
fix: lookup pkdId using nodeId as they are different

### PR DESCRIPTION
**Pull request by aarlaud**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [x] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

Looks up the nodeId in the graph nodes list to then extract the pkgId accordingly since NodeId != PkgId.

### Notes for the reviewer

Does not change the behavior of the direct/indirect deps comparison, but collect the pkgId more reliably by getting the PkgId instead of assuming NodeId == PkgId, which doesn't hold true in some cases in graph generated by [this lib](https://github.com/snyk/dep-graph).

### CODE REVIEW

File: src/lib/snyk/dependencies.ts [hunk 1]
```diff
@@ -7,31 +7,56 @@ import { IssueWithPaths } from '../types'
...
         snykTestGraph = await depgraph.legacy.depTreeToGraph(snykDepsJsonResults as depgraph.legacy.DepTree, packageManager)
     }
     
     //const snykTestDeps = snykTestGraph.getPkgs()
     const snykTestDirectDepsNodeIDs = snykTestGraph.toJSON().graph.nodes[0].deps.map(dep => dep.nodeId)
-    const snykTestDirectDeps = snykTestGraph.toJSON().pkgs.filter(pkg => snykTestDirectDepsNodeIDs.indexOf(pkg.id) > 0)
-    const snykTestIndirectDeps = snykTestGraph.toJSON().pkgs.filter(pkg => snykTestDirectDepsNodeIDs.indexOf(pkg.id) <= 0)
+    const snykTestDirectDepsPkgIDs: Array<string> = []
+    
+        snykTestDirectDepsNodeIDs.forEach(nodeId => {
```
Comment (lili2311) [inline src/lib/snyk/dependencies.ts:20]: is there a test you can update for this to show it working?

Comment (aarlaud) [reply src/lib/snyk/dependencies.ts:20]: added test case

File: src/lib/snyk/dependencies.ts [hunk 2]
```diff
@@ -7,31 +7,56 @@ import { IssueWithPaths } from '../types'
...
+            const node = snykTestGraph.toJSON().graph.nodes.find(node => node.nodeId == nodeId)
```
Comment (lili2311) [inline src/lib/snyk/dependencies.ts:21]: This returns the first match only, is it ever possible to have multiple matches?

File: src/lib/snyk/dependencies.ts [hunk 3]
```diff
@@ -7,31 +7,56 @@ import { IssueWithPaths } from '../types'
...
+            } else {
+                snykTestDirectDepsPkgIDs.push(node.pkgId)
+            }
+            
+        })
+    
     
+
+    const snykTestDirectDeps = snykTestGraph.toJSON().pkgs.filter(pkg => snykTestDirectDepsPkgIDs.indexOf(pkg.id) > 0)
+    const snykTestIndirectDeps = snykTestGraph.toJSON().pkgs.filter(pkg => snykTestDirectDepsPkgIDs.indexOf(pkg.id) <= 0)
```
Comment (lili2311) [inline src/lib/snyk/dependencies.ts:33]: Also a nice alternative is `snykTestDirectDepsPkgIDs.includes(pkg.id)`

File: src/lib/snyk/dependencies.ts [hunk 4]
```diff
@@ -7,31 +7,56 @@ import { IssueWithPaths } from '../types'
...
     
     const snykProjectGraph = depgraph.createFromJSON(monitoredProjectDepGraph.depGraph as depgraph.DepGraphData)
-    //const snykProjectDeps = snykTestGraph.getPkgs()
+
     const snykProjectDirectDepsNodeIDs = snykProjectGraph.toJSON().graph.nodes[0].deps.map(dep => dep.nodeId)
```
Comment (lili2311) [inline src/lib/snyk/dependencies.ts:37]: do we need to do this again, there is `snykTestDirectDepsNodeIDs` already?

File: test/lib/index-inline-with-proj-coord-dep-changes.test.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      'Added 1',
+      'cffi@1.14.5',
+      'Paths',
+      'cffi@1.14.5 no issue:',
+      'cryptography@3.3.2=>cffi@1.14.5',
+      'psycopg2cffi@2.8.1=>cffi@1.14.5',
+      'social-auth-core@3.3.3=>cryptography@3.3.2=>cffi@1.14.5',
+      'social-auth-app-django@4.0.0=>social-auth-core@3.3.3=>cryptography@3.3.2=>cffi@1.14.5',
+      'Removed 1',
+      'No new issues found !',
```
Comment (lili2311) [inline test/lib/index-inline-with-proj-coord-dep-changes.test.ts:98]: `No new issues found !` is the space there on purpose? Should it be `No new issues found!`?

File: test/lib/index-inline-with-proj-coord-dep-changes.test.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    ];
+
+    //expect(consoleOutput).toContain('No new iss4ues found !');
```
Comment (lili2311) [inline test/lib/index-inline-with-proj-coord-dep-changes.test.ts:101]: can delete?

File: test/lib/index-inline-with-proj-coord-dep-changes.test.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    expectedOutput.forEach((line: string) => {
+      expect(consoleOutput.join()).toContain(line);
+    });
+    // => When testing, loaded as module therefore returning code === process.exitCode
+    expect(mockExit).toHaveBeenCalledWith(0);
+    //expect(result).toEqual(0);
```
Comment (lili2311) [inline test/lib/index-inline-with-proj-coord-dep-changes.test.ts:107]: can delete?

**Review by lili2311 — APPROVED** [review]

File: src/lib/snyk/dependencies.ts (hunk 2 above)
Comment (aarlaud) [reply src/lib/snyk/dependencies.ts:21]: yes it's fine, nodeId is meant to be unique as per depgraph property.

File: src/lib/snyk/dependencies.ts [hunk 8]
```diff
@@ -7,31 +7,56 @@ import { IssueWithPaths } from '../types'
...
     let snykTestGraph: depgraph.DepGraph
     if(snykDepsJsonResults && snykDepsJsonResults.depGraph){ // Getting graph
+        
         snykTestGraph = depgraph.createFromJSON(snykDepsJsonResults.depGraph)
     } else { // Getting legacy dep tree
         snykTestGraph = await depgraph.legacy.depTreeToGraph(snykDepsJsonResults as depgraph.legacy.DepTree, packageManager)
     }
     
     //const snykTestDeps = snykTestGraph.getPkgs()
     const snykTestDirectDepsNodeIDs = snykTestGraph.toJSON().graph.nodes[0].deps.map(dep => dep.nodeId)
```
[the remark this answers is not in the document]
Comment (aarlaud) [reply src/lib/snyk/dependencies.ts:17]: it shouldn't happen, we always should have a root node.

File: src/lib/snyk/dependencies.ts (hunk 4 above)
Comment (aarlaud) [reply src/lib/snyk/dependencies.ts:37]: yes, here it's the direct dep nodeIds from the monitored snapshot coming via the API.

File: test/lib/index-inline-with-proj-coord-dep-changes.test.ts (hunk 5 above)
Comment (aarlaud) [reply test/lib/index-inline-with-proj-coord-dep-changes.test.ts:98]: no clue, what's the proper way? no space?

File: src/lib/snyk/dependencies.ts (hunk 3 above)
Comment (aarlaud) [reply src/lib/snyk/dependencies.ts:33]: I'll need to check later. I agree but simply swapping this out now breaks things. Probably missing something.

**Pull request merged**